app/knights/knight.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from app.knights.inventory.weapon import Weapon
+from app.knights.inventory.armour import Armour
+from app.knights.inventory.consumables import Potion
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.protection = 0
+        self.weapon = Weapon(knight["weapon"])
```

Comment:
Creating instances inside of `__init__` is possible but it makes logic of the class quite complicated. Is a good practice to break down complex logic into smaller pieces, like here you could create instances of inventory classes elsewhere and simply pass created instances to `__init__`
```
class Knight:

    def __init__(
            self,
            name: str,
            power: int,
            hp: int,
            weapon: Weapon,
            armour: Optional[list[Armour]] = None,
            potion: Optional[Potion] = None,
    ) -> None:
        self.name = name
        self.power = power
        self.hp = hp
        self.protection = 0
        self.armour = armour or []
        self.weapon = weapon
        self.potion = potion